U/USalign/build_tarballs.jl

```diff
@@ -0,0 +1,89 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "USalign"
+version = v"2022.9.24"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/pylelab/USalign.git", "944c6a81c453b43d173d8f18ff3197d59650c2cc")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd USalign/
+# Remove the -ffast-math flag from the suggested g++ command as required by BinaryBuilder.
+sed -i 's/-ffast-math//g' Makefile
+# The following sed statements are needed to support Linux powerpc64le {libc=glibc}.
+if [[ "${target}" == powerpc64le* ]]; then
+    sed -i 's/\bA0\b/A0_var/g' TMalign.h;
+    sed -i 's/\bB0\b/B0_var/g' TMalign.h;
+    sed -i 's/\bC0\b/C0_var/g' TMalign.h;
+    sed -i 's/\bD0\b/D0_var/g' TMalign.h;
+fi
+# Remove the -static flag to support macOS aarch64 and x86_64.
+if [[ "${target}" == *-apple-darwin* ]]; then
+    sed -i 's/-static//g' Makefile;
+fi
+# Add a dummy pstream.h to allow compilation on Windows.
+if [[ "${target}" == *-mingw* ]]; then
+    echo -e '#ifndef PSTREAM_H\n' \
+    '#define PSTREAM_H\n' \
+    '\n#include <iostream>\n' \
+    '\nnamespace redi {\n' \
+    '    class ipstream {\n' \
+    '    public:\n' \
+    '        ipstream() {\n' \
+    '            std::cout << "POSIX is not supported on Linux, so this program cannot deal with gzipped files on this platform.\\n";\n' \
+    '        }\n' \
+    '\n        void open(const std::string&) {\n' \
+    '            std::cout << "ipstream::open is not supported on this platform.\\n";\n' \
+    '        }\n' \
+    '        bool good() const { return false; }\n' \
+    '        void close() {}\n' \
+    '        friend ipstream& getline(ipstream& is, std::string& str) { return is; }\n' \
+    '    };\n' \
+    '}\n' \
+    '\n#endif // PSTREAM_H' > pstream.h;
```

Comment:
Do you know [here documents](https://tldp.org/LDP/abs/html/here-docs.html)?  Also, it may be easier to add the the header file as an [extra source](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.DirectorySource) and copy the file where you need it.  Where does it come from BTW?

Reply:
Thanks! :) I didn't know about the extra source possibility. I will use the _here document_ as the header is on UNIX. This is only a dummy file I created defining the needed methods to avoid the compilation failing on Windows where POSIX and pstream are not supported (as far as I understood).